### PR TITLE
User should be notified of an outbid or (any non-winning status) when attempting to confirm a bid

### PR DESCRIPTION
**Pull request by yuki24**
addresses https://artsyproduct.atlassian.net/browse/AUCT-713

## Todos

 * [x] Add more test coverage
 * [x] Add screenshots for cases where:
    * `OUTBID`
    * `RESERVE_NOT_MET`
    * `SALE_CLOSED`
    * Javascript error (e.g. disable WiFi and try bidding)

### `OUTBID` or `RESERVE_NOT_MET`

AUCT-713-outbid

### `SALE_CLOSED`

AUCT-713-lot-closed

### Javascript error

AUCT0-713-js-error

### CODE REVIEW

File: src/Apps/Auction/Routes/ConfirmBid/BidderPositionQuery.tsx [hunk 1]
```diff
@@ -2,14 +2,23 @@ import {
   BidderPositionQueryResponse,
   BidderPositionQueryVariables,
 } from "__generated__/BidderPositionQuery.graphql"
-import { graphql } from "react-relay"
-import { Environment, fetchQuery } from "relay-runtime"
+import { graphql, GraphQLTaggedNode } from "react-relay"
+import { CacheConfig, Environment, fetchQuery, Variables } from "relay-runtime"
+
+// @types/relay-runtime 1.3.5 does not have the correct signature so making our
```
Comment (erikdstock) [inline src/Apps/Auction/Routes/ConfirmBid/BidderPositionQuery.tsx]: 🤙

File: src/Apps/Auction/Routes/ConfirmBid/index.tsx [hunk 2]
```diff
@@ -45,20 +49,15 @@ export const ConfirmBidRoute: React.FC<ConfirmBidProps> = props => {
...
-        relay.environment,
-        {
-          onCompleted: data => {
-            resolve(data)
-          },
-          onError: error => {
-            reject(error)
-          },
+    return new Promise<ConfirmBidCreateBidderPositionMutationResponse>(
+      async (resolve, reject) => {
```
Comment (erikdstock) [inline src/Apps/Auction/Routes/ConfirmBid/index.tsx]: is this mix of Promise and `async` necessary?

File: src/Apps/Auction/Routes/ConfirmBid/index.tsx [hunk 3]
```diff
@@ -89,31 +88,16 @@ export const ConfirmBidRoute: React.FC<ConfirmBidProps> = props => {
...
-    } else if (error.message) {
-      errorMessages = [error.message]
-    }
-
-    trackConfirmBidFailed(bidderId, errorMessages)
-
+    trackConfirmBidFailed(bidderId, [error.message])
     actions.setSubmitting(false)
-    actions.setStatus("submissionFailed")
+    // actions.setStatus("submission Failed.")
```
Comment (erikdstock) [inline src/Apps/Auction/Routes/ConfirmBid/index.tsx]: Maybe this should be deleted/not sure if it is still necessary.

File: src/Apps/Auction/Routes/ConfirmBid/index.tsx (hunk 2 above)
Comment (yuki24) [reply src/Apps/Auction/Routes/ConfirmBid/index.tsx]: Removed, things seem to be working without it.

File: src/Apps/Auction/Routes/ConfirmBid/index.tsx (hunk 3 above)
Comment (yuki24) [reply src/Apps/Auction/Routes/ConfirmBid/index.tsx]: It seems like this is how Formik keeps track of error state, so I brought it back and used it to show an error message.

**Comment by yuki24** [comment]
Alright, this is now ready for review.

**Pull request merged**